src/model/transaction/randomizeBlockMapKeys.js

```diff
@@ -85,50 +69,67 @@ const randomizeContentBlockNodeKeys = (blockMap: BlockMap): BlockMap => {
             if (parentBlock) {
               const parentChildrenList = parentBlock.getChildKeys();
               blockMapState.setIn(
-                parentKey,
-                parentBlock.merge({
-                  children: parentChildrenList.set(
-                    parentChildrenList.indexOf(block.getKey()),
-                    key,
-                  ),
-                }),
+                [parentKey, 'children'],
+                parentChildrenList.set(
+                  parentChildrenList.indexOf(block.getKey()),
+                  key,
+                ),
               );
             } else {
-              blockMapState.mergeIn(
-                oldKey,
-                block.merge({
-                  parent: null,
-                }),
+              // - no parentblock means this will become a rootBlock so we need to update the
+              // lastRootBlock nextSibling link
+              if (lastRootBlock) {
+                blockMapState.setIn(
+                  [lastRootBlock.getKey(), 'nextSibling'],
+                  key,
+                );
+
+                // TODO
+                // how can we update this blocks prevSibling with the newly randomized key
+                // of the prevSibling block ??
```

Comment:
same as above

| Remove this todos since they are no longer applicable, and replace with something that describe the solution for this previous behaviour